### PR TITLE
Handle failure to access API endpoint

### LINKED ISSUE #22: Failure to access API endpoint is not handled properly
If the endpoint is not accessible when generating a new project, the wizard just waits endlessly despite an error being reported in the logs:
```
rejected promise not handled within 1 second: StatusCodeError: 404 - "RESTEASY003210: Could not find resource for full path: http://quarkus-generator.6923.rh-us-east-1.openshiftapps.com/extensions"
extensionHostProcess.js:759
stack trace: StatusCodeError: 404 - "RESTEASY003210: Could not find resource for full path: http://quarkus-generator.6923.rh-us-east-1.openshiftapps.com/extensions"
```

### PR DESCRIPTION
**Pull request by xorye**
Fixes #22

Here are the two error messages:

When the extensions list cannot be reached:
image

When the zip cannot be downloaded and opened:
image

When getting the extensions, if the api url hangs, there is a timeout [here](https://github.com/xorye/vscode-quarkus/blob/d5c8414da6230d5b7e5ce32682c8d8a30c0b36ca/src/utils/requestUtils.ts#L26) to take care of that.

Signed-off-by: David Kwon <dakwon@redhat.com>

### CODE REVIEW

File: src/utils/requestUtils.ts [hunk 1]
```diff
@@ -14,17 +14,20 @@
...
 import { QExtension, APIExtension } from '../definitions/extension';
-import { State } from '../definitions/state';
+import { ProjectGenState } from '../definitions/projectGenerationState';
 
 export async function getQExtensions(apiUrl: string): Promise<QExtension[]> {
-
-  return await request.get(`${apiUrl}/extensions`)
+  const requestOptions = {
+    uri: `${apiUrl}/extensions`,
+    timeout: 1500
```
Comment (fbricon) [inline src/utils/requestUtils.ts]: 30000ms is more realistic

Comment (xorye) [reply src/utils/requestUtils.ts]: changed to 30000ms

File: src/generateProject/pickExtensions.ts [hunk 2]
```diff
@@ -40,21 +40,28 @@ interface QuickPickItem {
...
 
   const apiUrl = settings.apiUrl ? settings.apiUrl : DEFAULT_API_URL;
 
-  const allExtensions: QExtension[] = await getQExtensions(apiUrl);
+  let allExtensions: QExtension[];
+
+  try {
+    allExtensions = await getQExtensions(apiUrl);
+  } catch (err) {
+    state.wizardInterrupted = {reason: 'Unable to reach API endpoint for the extensions list.'};
```
Comment (fbricon) [inline src/generateProject/pickExtensions.ts]: 'Unable to reach '+ apiUrl +'/extensions`

**Review by fbricon — CHANGES_REQUESTED** [review]

**Comment by xorye** [comment]
@fbricon I made the changes.

**Comment by xorye** [comment]
Conflicts are fixed

**Comment by fbricon** [comment]
Thanks @xorye !

**Pull request merged**